istioctl/pkg/dashboard/dashboard.go

```diff
@@ -411,23 +416,27 @@ func istioDebugDashCmd(ctx cli.Context) *cobra.Command {
   istioctl d istiod-debug pilot-123-456.istio-system
 `,
 		RunE: func(c *cobra.Command, args []string) error {
-			if labelSelector == "" && len(args) < 1 {
+			if labelSelector == "" && opts.Revision == "" && len(args) < 1 {
 				c.Println(c.UsageString())
-				return fmt.Errorf("specify a pod or --selector")
+				return fmt.Errorf("specify a pod, --selector, or --revision")
 			}
 
-			if labelSelector != "" && len(args) > 0 {
+			if (labelSelector != "" && len(args) > 0) || (labelSelector != "" && opts.Revision != "") || (len(args) > 0 && opts.Revision != "") {
 				c.Println(c.UsageString())
-				return fmt.Errorf("name cannot be provided when a selector is specified")
+				return fmt.Errorf("only one of name, --selector, or --revision can be specified")
 			}
-
 			client, err := ctx.CLIClientWithRevision(opts.Revision)
 			if err != nil {
 				return fmt.Errorf("failed to create k8s client: %v", err)
 			}
 
 			var podName, ns string
+			if opts.Revision != "" {
+				labelSelector = "istio.io/rev=" + opts.Revision + ", app=istiod"
+			}
+
 			if labelSelector != "" {
+				labelSelector += ", app=istiod"
```

Comment:
I'm not strictly opposed to this, I just don't understand why we would need it now, but not before

Reply:
The developers probably didn't consider the fixed "app=istiod" tag of istiod.

Reply:
This actually has nothing to do with the `--revision` flag, it's just an additional small improvement

Reply:
I see. Could you change it to `app.kubernetes.io/name` then